source/screenExplorer.py

```diff
@@ -1,8 +1,69 @@
+#screenExplorer.py
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2012-2018 NV Access Limited, Joseph Lee
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+"""Module which implements various utilities for screen exploration.
+This includes touchscreen movement as well as handling coordinate announcement for mouse and objects.
+"""
+
 import api
 import tones
 import controlTypes
 import textInfos
 import speech
+import screenBitmap
+import config
+import mouseHandler
+import wx
+
+def playLocationCoordinates(x, y, screenWidth, screenHeight, screenMinPos=None, detectBrightness=True,blurFactor=0):
+	"""Plays a tone indicating the XY-coordinate for the given location.
+	This works for keyboard-based object navigation/focus movement, mouse movement and during touch hover gesture.
+	Left to right adjusting the volume between left and right speakers.
+	Top to bottom adjusts the pitch of the sound.
+	Brightness adjusts the volume of the sound (applicable for mouse movement).
+	Coordinates (x, y) are absolute, and can be negative.
+	"""
+	# Have a fake wx.Point in case min pos is not defined on multiple monitors.
+	if screenMinPos is None:
+		screenMinPos = wx.Point()
+	# Do not play any tone if offscreen.

```

Comment:
The old version of this function had:

```
	# make relative to (0,0) and positive
	x = x - screenMinPos.x
	y = y - screenMinPos.y
```

Presumably this is still necessary? I believe that with multiple screens, negative x and y values are expected. `ScreenMinPos` is intended to allow these negative values to be normalized.